### ISSUE TITLE
Connection Builders

### ISSUE THREAD

**Issue by jen20** [jen20/EventStore.Java#14]
A fluent builder should be supported, similarly to the .NET client which exposes the following settings: http://docs.geteventstore.com/dotnet-api/3.0.2/connecting-to-a-server/

**Comment by jen20** [comment]
These have been added in 1867cbf (for settings)

**Comment by jen20** [comment]
This now works very differently to the .NET client. The following code samples show the building of entire connections via a fluent API:

``` java
    public ValidateAPIDesign() {
        SubscribableEventStoreConnection connection = EventStoreConnection.newClusterConnectionBuilder()
                .withConnectionName("subscriptions")
                .discoverClusterViaDNS("cluster.local", 30778)
                .withDefaultCredentials(new Credentials("test.user", new char[] { 'p', 'a', 's', 's'}))
                .withUnlimitedReconnectionAttempts()
                .build();

        ReadableEventStoreConnection connection2 = EventStoreConnection.newSingleNodeBuilder()
                .withConnectionName("singlenode")
                .withConnectionAddress(new InetSocketAddress("127.0.0.1", 2113))
                .withUnlimitedReconnectionAttempts()
                .build();

    }
```

**Comment by jen20** [comment]
Fixed in 467d413
